staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -508,6 +508,87 @@ func (r *Request) URL() *url.URL {
 	return finalURL
 }
 
+// finalURLTemplate is similar to URL(), but will make all specific parameter values equal
+// - instead of name or namespace, "{name}" and "{namespace}" will be used, and all query
+// parameters will be reset. This creates a copy of the url so as not to change the
+// underlying object.
+func (r Request) finalURLTemplate() url.URL {
+	newParams := url.Values{}
+	v := []string{"{value}"}
+	for k := range r.params {
+		newParams[k] = v
+	}
+	r.params = newParams
+	u := r.URL()
+	if u != nil {
```

Comment:
```suggestion
	if u == nil {
```

Reply:
and you will be able to remove the check at https://github.com/kubernetes/kubernetes/blob/3186c7390667ee065ffa73f1890c7cf51bc617e8/staging/src/k8s.io/client-go/rest/request.go#L531 to not make it redundant

Reply:
I totally miss this. Thanks